src/esp/nav/PathFinder.cpp

```diff
@@ -132,9 +132,9 @@ class IslandSystem {
                   const dtQueryFilter* filter,
                   const uint32_t newIslandId,
                   const dtPolyRef& startRef,
-                  std::vector<vec3f>& islandVerts) {
+                  std::vector<vec3f>* islandVerts) {
```

Comment:
This is in direct contrast with what Magnum's coding guidelines have (and I'm well aware of this Google coding style rule and have a different opinion):

- In Magnum, a pointer indicates a "nullable" parameter (whereas here passing a `nullptr` would crash), while a reference indicates a parameter that can't be null. This is in order to move the null-checking responsibility to the caller of the function, not the function itself (where, in order to be safe, it would need a lot of `assert(islandVerts)` noise everywhere). A consistency in this particular rule is crucial, otherwise you can't trust the API to accept `nullptr` in pointer parameters.
- If a function has "out" parameters, it's explicitly said so in the documentation as `@param[out]` (for example, [MeshTools::flipNormals()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1MeshTools.html#af4417b043b3de224d43edf1748f880fc))

Can this particular rule be disabled? Because it'll inevitably lead to Magnum-inspired APIs being *almost* the same, except for `&` parameters being converted to `*`.

Reply:
> This is in direct contrast with what Magnum's coding guidelines have (and I'm well aware of this Google coding style rule and have a different opinion):
> 
> * In Magnum, a pointer indicates a "nullable" parameter (whereas here passing a `nullptr` would crash), while a reference indicates a parameter that can't be null. This is in order to move the null-checking responsibility to the caller of the function, not the function itself (where, in order to be safe, it would need a lot of `assert(islandVerts)` noise everywhere). A consistency in this particular rule is crucial, otherwise you can't trust the API to accept `nullptr` in pointer parameters.
> * If a function has "out" parameters, it's explicitly said so in the documentation as `@param[out]` (for example, [MeshTools::flipNormals()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1MeshTools.html#af4417b043b3de224d43edf1748f880fc))
> 
> Can this particular rule be disabled? Because it'll inevitably lead to Magnum-inspired APIs being _almost_ the same, except for `&` parameters being converted to `*`.
> 
> EDIT: just saw the comment from @erikwijmans -- I guess we agree on this one :)

I think there are pros and cons to both approaches. Avoiding a null-check is  nice. Being able to easily reason about mutability in the context of the call is also nice. I slightly prefer the former but not strongly. However, I think consistency with Magnum is an even more important concern.

I will remove this check. I will try to add a check for the Magnum convention in the future.

Reply:
> References nicely deal with ownership

> Raw pointers should not be used to transfer ownership. To transfer ownership, a smart pointer should be used.

I think you're both saying the same thing, just differently :) If a function takes a reference, it's clear it *does not* take ownership (nobody calling `delete` on it and such). On the other hand, some 3rd APIs might be designed pre-C++11 (so no smart pointers) and C APIs have just pointers, which means if there's a pointer, it's less clear if there's some ownership transfer happening or not.